lib/node_modules/@stdlib/napi/argv-complex128/src/main.c

```diff
@@ -0,0 +1,69 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/napi/argv_complex128.h"
+#include "stdlib/assert/napi/status_ok.h"
+#include "stdlib/assert/napi/is_type.h"
+#include "stdlib/complex/float64.h"
+#include <node_api.h>
+
+/**
+* Converts a Node-API value to a double-precision complex floating-point number.
+*
+* @param env       environment under which the function is invoked
+* @param value     Node-API value
+* @param out       destination for storing output value
+* @param message   error message
+* @param err       pointer for storing a JavaScript error
+* @return          status code indicating success or failure (returns `napi_ok` if success)
+*
+* @example
+* #include "stdlib/napi/argv_complex128.h"
+* #include "stdlib/complex/float64.h"
+* #include <node_api.h>
+*
+* static napi_value addon( napi_env env, napi_callback_info info ) {
+*     napi_value value;
+*
+*     // ...
+*
+*     stdlib_complex128_t out;
+*     napi_value err;
+*     napi_status status = stdlib_napi_argv_complex128( env, value, &out, "Must be a Complex128 instance.", &err );
+*     assert( status == napi_ok );
+*     if ( err != NULL ) {
+*         assert( napi_throw( env, err ) == napi_ok );
+*         return NULL;
+*     }
+*
+*     // ...
+* }
+*/
+napi_status stdlib_napi_argv_complex128( const napi_env env, const napi_value value, stdlib_complex128_t *out, const char *message, napi_value *err ) {
+	double real;
+	double imaginary;
+
+	stdlib_assert_napi_value_is_type( env, value, napi_object, message, err );
+	if ( *err != NULL ) {
+		return napi_ok;
+	}
+	STDLIB_ASSERT_NAPI_STATUS_OK_RET_VALUE( env, napi_get_property( env, value, re, &real ), "", napi_ok )
+	STDLIB_ASSERT_NAPI_STATUS_OK_RET_VALUE( env, napi_get_property( env, value, im, &imaginary ), "", napi_ok )
+	*out = stdlib_complex128( real, imaginary );
```

Comment:
This can be improved. E.g., see https://github.com/stdlib-js/stdlib/blob/615bcf165b0a9a9aad5f6591024e2dd1655c8914/lib/node_modules/%40stdlib/math/base/napi/binary/src/main.c#L168 for reference.

1. You state that the input value must be a Complex128 instance. We should relax that requirement. We should allow any object coming from JS with a real and imaginary component (may be either an "own" or "inherited" property).
2. You don't validate that the real and imaginary components which are numeric.
3. You need to unwrap the real and imaginary components before passing to `stdlib_complex128`. E.g., https://github.com/stdlib-js/stdlib/blob/615bcf165b0a9a9aad5f6591024e2dd1655c8914/lib/node_modules/%40stdlib/math/base/napi/binary/src/main.c#L268.